apps/prairielearn/src/pages/administratorQuery/administratorQuery.ts

```diff
@@ -82,33 +114,36 @@ router.post(
       encoding: 'utf8',
     });
 
-    const queryParams = _.pick(
-      req.body,
-      _.map(info.params || {}, (p) => p.name),
-    );
-    const params = {
+    const queryParams = {};
+    if (info.params) {
+      info.params.forEach((p) => {
+        queryParams[p.name] = req.body[p.name];
+      });
+    }
+
+    const params: AdministratorQueryRunParams = {
       name: req.params.query,
       sql: querySql,
       params: JSON.stringify(queryParams),
       authn_user_id: res.locals.authn_user.user_id,
       error: null,
       result: null,
+      formatted_date: null,
     };
     try {
       const result = await sqldb.queryAsync(querySql, queryParams);
-      params.result = JSON.stringify({
+      params.result = {
         rowCount: result.rowCount,
-        columns: _.map(result.fields, (f) => f.name),
+        columns: result.fields.map((f) => f.name),
         rows: result.rows,
-      });
+      };
     } catch (err) {
       params.error = err.toString();
     }
-
     const result = await sqldb.queryOneRowAsync(sql.insert_query_run, params);
```

Comment:
Ditto on using `queryRow` for validation.

Reply:
Resolved in commit 2e8fb29.